torch/nn/parallel/distributed.py

```diff
@@ -660,10 +660,11 @@ def forward(self, *inputs, **kwargs):
             self._check_global_requires_backward_grad_sync(is_joined_rank=False)
 
         if self.device_ids:
-            inputs, kwargs = self.scatter(inputs, kwargs, self.device_ids)
             if len(self.device_ids) == 1:
+                inputs, kwargs = self.to_kwargs(inputs, kwargs, self.device_ids[0])
                 output = self.module(*inputs[0], **kwargs[0])
```

Comment:
Curious, any reason we still need to return `inputs` and `kwargs` as a list?

Reply:
I was mostly doing that to keep parity with the current version, but we can probably remove it in this code path.